client/asset/dcr/dcr.go

```diff
@@ -937,13 +942,15 @@ func (dcr *ExchangeWallet) Swap(swaps *asset.Swaps) ([]asset.Receipt, asset.Coin
 	}
 
 	// Add change, sign, and send the transaction.
+	dcr.fundingMtx.Lock()         // before generating change output
+	defer dcr.fundingMtx.Unlock() // hold until after returnCoins and lockFundingCoins(change)
```

Comment:
This was a bug that snuck in with the split tx commit - `lockFundingCoins` was called below without locking the mutex.